### PR TITLE
feat: parse solution text with preserve order

### PR DESCRIPTION
**Pull request by rayzhou-bit**
For the XMLParser, the `preserveOrder` option needs to be set to true for us to capture solutions properly when there are nested tags. Otherwise, the parser will bundle certain texts together in an undesired fashion.

https://2u-internal.atlassian.net/browse/TNL-10728

### CODE REVIEW

File: src/editors/containers/ProblemEditor/data/OLXParser.js [hunk 1]
```diff
@@ -63,25 +63,27 @@ export const stripNonTextTags = ({ input, tag }) => {
 export class OLXParser {
   constructor(olxString) {
     this.problem = {};
-    this.questionData = {};
-    const questionOptions = {
+    this.problemArray = [];
+    const parserOptions = {
```
Comment (connorhaugh) [inline src/editors/containers/ProblemEditor/data/OLXParser.js:67]: there should be a comment here explaining the three different options and why they are different

File: src/editors/containers/ProblemEditor/data/OLXParser.js [hunk 2]
```diff
@@ -366,12 +367,12 @@ export class OLXParser {
         questionArray.push(tag);
       } else if (responseKeys.includes(tagName)) {
         /* <label> and <description> tags often are both valid olx as siblings or children of response type tags.
-         They, however, do belong in the question, so we append them to the question.
+          They, however, do belong in the question, so we append them to the question.
```
Comment (connorhaugh) [inline src/editors/containers/ProblemEditor/data/OLXParser.js:370]: You can remove this extra space

**Review by KristinAoki — CHANGES_REQUESTED** [review]
A lot of your changes have already been made to `main`. I would rebase with `main` and check the functionality.

**Comment by rayzhou-bit** [comment]
> A lot of your changes have already been made to `main`. I would rebase with `main` and check the functionality.

I tested the issue again and it is resolved in some ways. After a rebase attempt, the functionality that needs to be added is setting `trimValues` to false so the spaces before and after tags are kept in rich text fields.

**Comment by rayzhou-bit** [comment]
I'm going to create a new PR off of main and close this one out. I must have been working in an older one that matched prod behavior at the time.

**Pull request closed without merging**